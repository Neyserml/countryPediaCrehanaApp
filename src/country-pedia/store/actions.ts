interface DispatchObject<T, P = any> {
  type: T;
  payload?: P;
}

export enum CountryPediaAction {
  ClearState = "CLEAR_STATE",
  SetLanguages = "SET_LANGUAGES",
  SetCurrencies = "SET_CURRENCIES",
  SetRegions = "SET_REGIONS",
  SetCountries = "SET_COUNTRIES",
  SetCountryId = "SET_COUNTRY_ID",
}

export type CountryPediaDispatchObject<T> = DispatchObject<
  CountryPediaAction,
  T
>;
