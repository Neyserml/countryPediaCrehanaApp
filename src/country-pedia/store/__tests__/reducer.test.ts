import { CountryPediaAction } from "../actions";
import {
  initialState,
  countryPediaContextReducer as reducer,
} from "../reducer";

describe("CountryPedia context Reducer", () => {
  it("should clear state", () => {
    const state = {
      ...initialState,
      otherValue: 1,
    };

    const clearedState = reducer(state, {
      type: CountryPediaAction.ClearState,
    });

    expect(clearedState).toMatchSnapshot();
  });

  it("should throw error if dispatch invalid action type", () => {
    const type = "fakeActionType" as CountryPediaAction;
    try {
      reducer(initialState, {
        type,
      });
    } catch (error) {
      expect(error).toBeInstanceOf(Error);
      expect(error).toMatchSnapshot();
    }
  });

  it("should set countryId", () => {
    const state = {
      ...initialState,
    };

    const stateUpdated = reducer(state, {
      type: CountryPediaAction.SetCountryId,
      payload: "Peru",
    });

    expect(stateUpdated).toMatchSnapshot();
  });

  // so for all actions
});
