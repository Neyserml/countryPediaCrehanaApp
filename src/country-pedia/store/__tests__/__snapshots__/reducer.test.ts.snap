// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CountryPedia context Reducer should clear state 1`] = `
Object {
  "countries": null,
  "countryId": "",
  "currencies": null,
  "languages": null,
  "regions": null,
  "title": "CountryPedia",
}
`;

exports[`CountryPedia context Reducer should set countryId 1`] = `
Object {
  "countries": null,
  "countryId": "Peru",
  "currencies": null,
  "languages": null,
  "regions": null,
  "title": "CountryPedia",
}
`;

exports[`CountryPedia context Reducer should throw error if dispatch invalid action type 1`] = `[Error: Unknown countryPedia action type: fakeActionType]`;
